### PR TITLE
Add unit tests for com.blade.kit.CollectionKitTest

### PR DESCRIPTION
**Pull request by paulkennethkent**
I've analysed your code base and noticed that `blade.kit.CollectionKitTest` is not fully tested.

I've written some tests for the functions in this class with the help of [Diffblue Cover](https://www.diffblue.com/). Hopefully, these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

### CODE REVIEW

**Review by JohnLBergqvist — APPROVED** [review]

File: src/test/java/com/blade/kit/CollectionKitTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    final Object[] array = {};
+    final Object[] array2 = {null};
+    final ArrayList collection = new ArrayList();
+    final ArrayList collection2 = new ArrayList();
+    collection2.add(null);
+
+    Assert.assertTrue(CollectionKit.isEmpty(array));
+    Assert.assertFalse(CollectionKit.isEmpty(array2));
+    Assert.assertTrue(CollectionKit.isEmpty(collection));
+    Assert.assertFalse(CollectionKit.isEmpty(collection2));
```
Comment (freddytuxworth) [inline src/test/java/com/blade/kit/CollectionKitTest.java]: Add test for `isEmpty(null)`

File: src/test/java/com/blade/kit/CollectionKitTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    final Object[] array = {};
+    final Object[] array2 = {null};
+    final ArrayList collection = new ArrayList();
+    final ArrayList collection2 = new ArrayList();
+    collection2.add(null);
+
+    Assert.assertFalse(CollectionKit.isNotEmpty(array));
+    Assert.assertTrue(CollectionKit.isNotEmpty(array2));
+    Assert.assertFalse(CollectionKit.isNotEmpty(collection));
+    Assert.assertTrue(CollectionKit.isNotEmpty(collection2));
```
Comment (freddytuxworth) [inline src/test/java/com/blade/kit/CollectionKitTest.java]: Add test for `isNotEmpty(null)`

File: src/test/java/com/blade/kit/CollectionKitTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class CollectionKitTest {
+
+  @Test
+  public void isEmpty() {
+    final Object[] array = {};
```
Comment (peterschrammel) [inline src/test/java/com/blade/kit/CollectionKitTest.java:13]: Group the setup code with the respective assert:

```
    final Object[] array = {};
    Assert.assertTrue(CollectionKit.isEmpty(array));

    final Object[] array2 = {null};
    Assert.assertFalse(CollectionKit.isEmpty(array2));

    final ArrayList collection = new ArrayList();
    Assert.assertTrue(CollectionKit.isEmpty(collection));

    final ArrayList collection2 = new ArrayList();
    collection2.add(null);
    Assert.assertFalse(CollectionKit.isEmpty(collection2));
```

File: src/test/java/com/blade/kit/CollectionKitTest.java (hunk 1 above)
Comment (paulkennethkent) [reply src/test/java/com/blade/kit/CollectionKitTest.java]: `isEmpty(null)` results in a compile error because of an ambiguous method call but this works 

`final Object[] array3 = null;
  Assert.assertTrue(CollectionKit.isEmpty(array3));`

File: src/test/java/com/blade/kit/CollectionKitTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
+package com.blade.kit;
+
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.HashMap;
+import org.junit.Assert;
```
Comment (sudo-jaa) [inline src/test/java/com/blade/kit/CollectionKitTest.java]: Provided this repo doesn't have any style requirement that prohibit this, I think it's preferable to import `org.junit.Assert.*;` and collapse your assertions down to, for example: `assertTrue`

Comment (peterschrammel) [reply src/test/java/com/blade/kit/CollectionKitTest.java]: You could still use import static to get rid of Assert.

**Review by peterschrammel — APPROVED** [review]
